### ISSUE TITLE
Column header IDs duplicated across tables

### ISSUE THREAD

**Issue by arbelt** [rstudio/gt#1247]
## Prework

* [x] Read and agree to the [code of conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html) and [contributing guidelines](https://github.com/rstudio/gt/blob/master/.github/CONTRIBUTING.md).
* [x] If there is [already a relevant issue](https://github.com/rstudio/gt/issues), whether open or closed, comment on the existing thread instead of posting a new issue.

## Description

Column header IDs are based only on the column, so a document with multiple tables that each have a "Total" column will result in 2 `th` elements being given the id `Total`.

## Reproducible example

````
---
title: "Untitled"
format: html
---


```{r}
library(gt)
```

```{r}
gt(cars)
```

```{r}
gt(cars)
```

````

## Expected result

There should be no duplicated IDs. But here the first and second tables both have `th` elements with id `speed` and `dist`.

According to https://developer.mozilla.org/en-US/docs/Learn/HTML/Tables/Advanced, `scope` should be enough for accessibility purposes, so perhaps provide an option to disable assigning header and row IDs? Otherwise, I'd expect the header IDs to be prefixed by the table id.

## Session info

```
R version 4.2.2 (2022-10-31)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] gt_0.8.0.9000   lubridate_1.9.2 forcats_1.0.0   stringr_1.5.0   dplyr_1.1.0     purrr_1.0.1     readr_2.1.4     tidyr_1.3.0     tibble_3.1.8   
[10] ggplot2_3.4.1   tidyverse_2.0.0

loaded via a namespace (and not attached):
 [1] dbplyr_2.3.1            pillar_1.8.1            compiler_4.2.2          oira.datasets_0.1.0.900 tools_4.2.2             digest_0.6.31          
 [7] timechange_0.2.0        lifecycle_1.0.3         gtable_0.3.1            pkgconfig_2.0.3         rlang_1.0.6             box_1.1.2.9000         
[13] DBI_1.1.3               cli_3.6.0               rstudioapi_0.14         commonmark_1.8.1        yaml_2.3.7              haven_2.5.2            
[19] xfun_0.37               fastmap_1.1.1           duckdb_0.7.1-1          withr_2.5.0             knitr_1.42              sass_0.4.5             
[25] generics_0.1.3          fs_1.6.1                vctrs_0.5.2             hms_1.1.2               grid_4.2.2              tidyselect_1.2.0       
[31] glue_1.6.2              R6_2.5.1                fansi_1.0.4             blob_1.2.3              tzdb_0.3.0              magrittr_2.0.3         
[37] htmltools_0.5.4         scales_1.2.1            ellipsis_0.3.2          labelled_2.10.0         colorspace_2.1-0        renv_0.17.0            
[43] utf8_1.2.3              stringi_1.7.12          munsell_0.5.0          
```

**Comment by arbelt** [comment]
Wanted to put in a ping on this since I noticed two related-ish issues recently: #1577 and #1490

IDs are supposed to be unique throughout a document, and as noted in #1577, not to contain whitespace [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id). The duplicated IDs consistently get flagged by any accessibility review. 

This continues to be a barrier to using gt for content intended for wider distribution. We work around this by parsing the output html, stripping out all the IDs, and re-saving, but it feels a little janky for a package that's as popular and polished as this one. This type of post-processing would also be a bit more prohibitive in a Shiny app.

**Comment by unh-iod** [comment]
@arbelt - 

I use the following to correct the HTML after building the table with gt.
This corrects spanner ids, column ids, and sets an attribute in the html tabindex="0" to allow keyboard nav using tab to highlight the table. Enjoy!

```{R}
# function to read in gt and render in more accessible html format
library(tidyverse)
library(xml2)
library(readr)

id_correction_fn <- function(tab_html,file = NULL){
  mod <- tab_html %>% 
    as_raw_html()%>%
    as.character()
  
  tmp1 <- data.frame(c = names(tab_html$`_data`),
                    n = nchar(names(tab_html$`_data`))) %>%
    .[-1,] %>% 
    arrange(desc(n))
  
  for (i in 1:length(tmp1$n)){
    mod <- mod %>% 
      str_replace_all(paste('\\Q',tmp1$c[i],'\\E','"',sep=""),
                      paste("col_id_",i,'"',sep="")) 
  }
  tmp2 <- data.frame(s = tab_html$`_spanners`$spanner_label %>% unlist(),
                    n = nchar(tab_html$`_spanners`$spanner_label)) %>%
    arrange(desc(n))
  
  if(length(tab_html$`_spanners`$spanner_label)>0){
    for (y in 1:length(tab_html$`_spanners`$spanner_label)){
      mod <- mod %>% 
        str_replace_all(paste('\\Q',tmp2$s[y],'\\E','"',sep=""),
                        paste("col_spanner_id_",y,'"',sep=""))
    }
  }
  mod <- mod %>% 
    str_replace_all('\\Q<table class="gt_table"\\E', '<table class="gt_table" tabindex="0"')
  
  mod %>%
    as_xml_document() %>%
    write_html(file %>% substr(1,30))
  
  cat(readr::read_file(file %>% substr(1,30)))
  
  invisible(file.remove(file %>% substr(1,30)))
}


# Example Call:
```
tab_html <- data %>% 
    gt(rowname_col = "Row1") |>
    tab_stubhead(label = "Row1")
  
id_correction_fn(tab_html,file = tlist$Title)

**Comment by unh-iod** [comment]
Also - based the title of this post, you may be using axe DevTools v4.76.0 - to review accessibility perhaps (?).  If so, the error it throws I have found to be typically inaccurate.  The error message referring to "header IDs duplicated across tables" is resolved when white space and special disallowed special characters are removed from all column ids.  Happy Coding!

**Comment by olivroy** [comment]
We made tweaks to html elements id in #1842, they still aren't unique, but feel free to open a PR if you have improvements to suggest
